pushsync/pusher.go

```diff
@@ -201,16 +190,28 @@ func (p *Pusher) sync() {
 					unsubscribe()
 				}
 
-				// delete from pushed items
-				for i := 0; i < len(syncedAddrs); i++ {
-					delete(p.pushed, syncedAddrs[i].Hex())
-				}
-
 				// set chunk status to synced, insert to db GC index
-				if err := p.store.Set(ctx, chunk.ModeSetSync, syncedAddrs...); err != nil {
+				if err := p.store.Set(ctx, chunk.ModeSetSyncPush, syncedAddrs...); err != nil {
 					log.Error("pushsync: error setting chunks to synced", "err", err)
 				}
 
+				// delete from pushed items
+				for i := 0; i < len(syncedAddrs); i++ {
+					addr := syncedAddrs[i]
+					item := p.pushed[addr.Hex()]
+					// increment synced count for the tag if exists
+					tag := item.tag
```

Comment:
at some point can we move this to the db too and eliminate tags from pusheditem completely?

Reply:
@zelig I spoke to @janos about this yesterday. I'd like to rename `pushIndex` to `tagAccessIndex` or something of this sort. `pushIndex` should be just `StoredTimestamp|Hash->nil`, `tagAccessIndex` should be `Hash->Tags`.
This way we decouple push from pull syncing, allow correct ownership of chunks to (possibly multiple) tags and don't reuse indexes that have different semantics for lookup purposes.
The only caveats would be over extra-incurred DB operations.

For your consideration.

Reply:
On another note: this is a bit more involved, requires changing a few things in localstore and requires proper testing and a migration for the DB (if we decide to migrate at all. we might get away with no migration at all)

Reply:
why we need to complicate stuff? And most importantly what is the issue?
Can I have a clear problem statement or evidence of a failed testcase?